### PR TITLE
allow custom handler 'changeDate' for inline

### PR DESCRIPTION
**Pull request by myname1614**
in this line overriding config param:
```
$this->clientEvents['changeDate'] = "function (e){ jQuery('#$id').val(e.format());}";
```

fix to allow add handler for this case:
```
<?= DatePicker::widget([
    'inline' => true,
    'clientEvents' => [
        'changeDate' => new \yii\web\JsExpression('function() {console.log("!!!")}'),
    ],
]) ?>

```

### CODE REVIEW

File: src/DatePicker.php [hunk 1]
```diff
@@ -107,12 +107,16 @@ public function registerClientScript()
 
         $options = !empty($this->clientOptions) ? Json::encode($this->clientOptions) : '';
 
+        $js[] = "$selector.datepicker($options);";
+
         if ($this->inline) {
+            if (!empty($this->clientEvents['changeDate'])) {
```
Comment (tonydspaniard) [inline src/DatePicker.php:113]: Shouldn't be?: 

```php 
if(empty($this->clientEvents['changeDate'])) {
    $this->clientEvents['changeDate'] = "function (e){ jQuery('#$id').val(e.format());}";
}
```

File: src/DatePicker.php [hunk 2]
```diff
@@ -107,12 +107,16 @@ public function registerClientScript()
...
+                $handler = $this->clientEvents['changeDate'];
+                $js[] = "$selector.on('changeDate', $handler);";
+            }
             $this->clientEvents['changeDate'] = "function (e){ jQuery('#$id').val(e.format());}";
         }
 
-        $js[] = "$selector.datepicker($options);";
-
         if (!empty($this->clientEvents)) {
```
Comment (tonydspaniard) [inline src/DatePicker.php:120]: The following statement does exactly what you did above

File: src/DatePicker.php (hunk 1 above)
Comment (myname1614) [reply src/DatePicker.php:113]: else //not empty
<input /> 
will not assigned value attr.
we lost your default handler

Comment (myname1614) [reply src/DatePicker.php:113]: if we set config  'inline' => true && changeDate = ...
js looks like:

```
;jQuery('#hidden_to').parent().datepicker({"autoclose":true,"format":"yyyy-mm-dd","startDate":"+1d"});
;jQuery('#hidden_to').parent().on('changeDate', function() {$("button", $(this).closest(".modal-dialog")).prop("disabled", false)});
;jQuery('#hidden_to').parent().on('changeDate', function (e){ jQuery('#hidden_to').val(e.format());});
```

File: src/DatePicker.php (hunk 2 above)
Comment (myname1614) [reply src/DatePicker.php:120]: I moved this line to top,
because $this->clientEvents array has duplicate keys "changeDate"

**Pull request closed without merging**